### ISSUE TITLE
No access to validate object?

### ISSUE THREAD

**Issue by leetercola** [bestguy/react-use-validated-state#3]
Trying to use date/time validator but it throws an errow saying you need to extend validate with a parse and format function. I tried passing both functions into the validationConstraints object to no avail. Ideas?

**Comment by bestguy** [comment]
Hmm, good question.  The examples only show setting on the validate object, which I think is a little messy to expose.

We use validate.single call internally, and I _thought_ this would work if I change the call signature:

```javascript
const date = useValidatedState('', {
    presence: true,
    datetime: true
  }, {
    parse: value => fecha.parse(value, 'M/D/YYYY'),
    format: value => fecha.format(value, 'M/D/YYYY')
  });
```

which could internally call:

`const validationMessage = validate.single(value, constraints, options);`

But does not seem to work, but looks like it should?:   
https://github.com/ansman/validate.js/blob/master/validate.js#L193

If you can find a good example of calling validate.single with datetime without using the validate.extend call would help.

**Comment by bestguy** [comment]
I opened question here, let's see if anyone replies with a solution: https://github.com/ansman/validate.js/issues/295

**Comment by bestguy** [comment]
Actually @leetercola , looks like a workaround is to import validate object somewhere in your code above the hook, and set the parse and format functions there.  It seems to work as a global:

```
import validate from 'validate.js';

validate.extend(validate.validators.datetime, {
  parse: value => fecha.parse(value, 'M/D/YYYY'),
  format: value => fecha.format(value, 'M/D/YYYY')
});
```

Kind of hackish but seems to work.

**Comment by leetercola** [comment]
Fair enough, I sort of assumed it might but like you said it feels dirty to do so

**Comment by leetercola** [comment]
Thanks for writing this, I like the concept a little bit more than a lot of the hooks for validation right now. They all are form dependent this seems a bit more modular and useful for state that may not be form driven

**Comment by bestguy** [comment]
Thanks, I appreciate it.  Yeah the goal is to "just work" without exposing the internal details. Validate.js comes close but I dislike this datetime situation.  I'll continue to look for a clean validation library option for this, as I also prefer the modular approach.
